Content.Server/GameObjects/Components/Sound/BaseEmitSoundComponent.cs

```diff
@@ -0,0 +1,27 @@
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Sound
+{
+    public enum EmitSoundMode
+    {
+        Auto,
+        Single,
+        RandomFromCollection
+    }
+
+    /// <summary>
+    /// Base sound emitter which defines most of the data fields.
+    /// Default EmitSoundMode is Auto, which will first try to play a sound collection,
+    /// and if one isn't assigned, then it will try to play a single sound.
+    /// </summary>
+    public abstract class BaseEmitSoundComponent : Component
```

Comment:
component inheritance is extremely sus but this doesn't look.. that bad, needs check by someone more experienced than me probably

Reply:
feels like it should just be an interface but idk if that works with datafields